1/backend/src/resolvers.js

```diff
@@ -1,67 +1,268 @@
+                `
+                const session = driver.session()
+                try {
+                    await session.run(createUserCypher)
+                } finally {
+                    await session.close()
+                }
+                return true
+            }
+            return false;
+        },
+        deleteTodo: async (parent, args, context) => {
+            if (args.id != null && context.token) {
+                const {driver} = context
+                const deleteTodoCypher = `
+                    MATCH (todo:Todo {id: $id})
+                    DELETE todo
+                `
+                const session = driver.session()
+                try {
+                    await session.run(deleteTodoCypher, {id: args.id})
                     return true
+                } finally {
+                    await session.close()
                 }
                 return false
             }
         },
-        editTodo: (parent, args, context) => {
-            if (args.id != null && args.message != null && context.token){
-                tod = data.todos.find(td => td.id === parseInt(args.id))
-                if (tod){
-                    tod.message = args.message;
-                    return tod;
+        editTodo: async (parent, args, context) => {
+            if (args.id != null && args.message != null && context.token) {
+                const {driver} = context
+                const editTodoCypher = `
+                    MATCH (todo:Todo {id: $id})
+                    SET todo.message = $message
+                    RETURN todo.id, todo.message, todo.completed
+                `
+                const session = driver.session()
+                try {
+                    const _response = await session.run(editTodoCypher, {id: args.id, message: args.message})
+                    const [response] = await _response.records.map(record => ({
+                        id: record.get('todo.id'),
+                        message: record.get('todo.message'),
+                        completed: record.get('todo.completed')
+                    }))
+                    if (response) {
+                        return response
+
+                    }
+                    return null
+                } finally {
+                    await session.close()
                 }
-                return
             }
 
         },
-        finishTodo: (parent, args, context) => {
-            if (args.id != null && context.token){
-                let tod = data.todos.find(td => td.id === parseInt(args.id));
-                if (tod){
-                    //Reverse status of item
-                    tod.completed = !tod.completed;
-                    return tod
+        finishWithMerge: async (parent, args, context) => {
+            if (args.id != null && context.token) {
+                const {driver} = context
+                const mergeTodoCypher = `
+                    MERGE (todo:Todo {id: $id})
```

Comment:
Why do you make a `MERGE` here? Do you ever want to create the todo with this resolver?